src/exchanges.js

```diff
@@ -12,6 +12,13 @@ var amqplib       = require('amqplib');
 var events        = require('events');
 var util          = require('util');
 var common        = require('./common');
+var taskcluster   = require('taskcluster-client');
+
+// wait 30 seconds before closing a channel, to allow pending operations to flush
+var CLOSE_DELAY = 30 * 1000;
+
+// unconditionally reconnect to pulse on this interval
+var RECONNECT_INTERVAL = "30 minutes"
 
```

Comment:
Why? I'm pretty sure rabbitmq only requires password when connecting...
Sure tc-pulse can break the connection, but doing so after ~30min seems a bit extreme...

Maybe breaking connections from tc-pulse every 48 hours, and in here every 36 hours... That way it'll never happen on heroku...

This is bound to have performance implications...